nav2_smac_planner/src/smac_planner_hybrid.cpp

```diff
@@ -534,19 +543,21 @@ nav_msgs::msg::Path SmacPlannerHybrid::createPlan(
     }
     _expansions_publisher->publish(msg);
 
-    // plot footprint path planned for debug
     if (_planned_footprints_publisher->get_subscription_count() > 0) {
+      // Clear all markers first
       auto marker_array = std::make_unique<visualization_msgs::msg::MarkerArray>();
+      visualization_msgs::msg::Marker clear_all_marker;
+      clear_all_marker.action = visualization_msgs::msg::Marker::DELETEALL;
+      marker_array->markers.push_back(clear_all_marker);
+      _planned_footprints_publisher->publish(std::move(marker_array));
```

Comment:
I added this to clear the previous footprints because as it was before, the footprints would just superimpose